### DISCUSSION TITLE
how to best aggregate data across components

### DISCUSSION THREAD

**Discussion by ccmsi** [vectordotdev/vector#14289] [Q&A] [answered]
vector validator doesnt recognize variables across components resulting in "error[E652]: only objects can be merged" despite in theory, this  would work. Eg when renaming / initialising .aggregator in test_remap, tapping input of test_remap shows .aggregator as initialized with {}. How should this best be dealt with ?

[transforms.test_init]
        type = "remap"
        inputs = ["test_file_in","test_socket_in"]
        source = """
.aggregator = {}
"""

[transforms.test_remap]
        type = "remap"
        inputs = ["test_init"]
        source = """
.aggregator |= {"x":1}
"""

**Comment by jszwedko** [comment, marked as answer]
The issue here is that `test_remap` doesn't know that `.aggregator` as an object as that type information isn't propagated across transforms (yet; this is something we are working on). You can do `.aggregator = merge!(.aggregator, {"x": 1})` instead, though, or:

```
.aggregator = object!(.aggregator)
.aggregator |= {"x":1}
```
